invader_payment_adyen_sepa/models/account_payment_order.py

```diff
@@ -119,6 +121,31 @@ def _invader_prepare_payment_transaction_data(self, acquirer_id):
             ]
         return list_vals
 
+    def open2generated(self):
+        self.ensure_one()
+        action = super().open2generated()
+        if (
```

Comment:
I would put that  condition in a method to centralize condition and avoid to repeat it along the code